app/main.py

```diff
@@ -1,34 +1,104 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self,
+                 row: int,
+                 column: int,
+                 is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple[int],
+                 end: tuple[int],
+                 is_drowned: bool = False) -> None:
+        if not (start[0] == end[0] or start[1] == end[1]):
+            raise ValueError("Decks shouldn't be located "
+                             "by diagonal)")
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        self.start = start
+        self.end = end
+        self.decks = {(i, j): Deck(i, j)
+                      for i in range(start[0], end[0] + 1)
+                      for j in range(start[1], end[1] + 1)}
+        self.is_drowned = is_drowned
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        return self.decks[(row, column)]
+
+    def fire(self, row: int, column: int) -> None:
+        self.get_deck(row, column).is_alive = False
+
+        if not any(deck.is_alive for deck in self.decks.values()):
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list[tuple]) -> None:
+        self.ships = {(x, y): Ship(x, y) for x, y in ships}
+        self.field = {}
+        self.field.update({deck: ship
+                           for ship in self.ships.values()
+                           for deck in ship.decks.keys()})
```

Comment:
Iterating over `dict.keys()` is the same as iterating over the dict.

Reply:
It's more readable when there is an indication that `decks` is a dictionary